app/assets/javascripts/components/projects/ProjectBoard.js

```diff
@@ -1,155 +1,202 @@
-      onClick: reverseColumns,
-      'data-id': 'reverse-button',
-      isVisible: projectBoard.reverse,
-      icon: 'fas fa-columns'
-    },
-    {
-      description: I18n.t('toggle_column', { column: 'chilly bin' }),
-      onClick: () => toggleColumn('chillyBin'),
-      'data-id': 'toggle-chilly-bin',
-      isVisible: projectBoard.visibleColumns.chillyBin,
-      icon: 'fas fa-snowflake'
-    },
-    {
-      description: I18n.t('toggle_column', { column: 'backlog' }),
-      onClick: () => toggleColumn('backlog'),
-      'data-id': 'toggle-backlog',
-      isVisible: projectBoard.visibleColumns.backlog,
-      icon: 'fas fa-th-list'
-    },
-    {
-      description: I18n.t('toggle_column', { column: 'done' }),
-      onClick: () => toggleColumn('done'),
-      'data-id': 'toggle-done',
-      isVisible: projectBoard.visibleColumns.done,
-      icon: 'fas fa-check-circle'
+
+    if (isSameColumn && sourceIndex === destinationIndex) {
+      return;
     }
-  ];
 
-    const columnProps = {
-      'data-id': projectBoard.reverse ? 'reversed-column' : 'normal-column',
-      columns: projectBoard.reverse ? columns.reverse() : columns
-    };
+    const newPosition = getNewPosition(
+      destinationIndex,
+      sourceIndex,
+      destinationArray,
+      isSameColumn,
+      dragStory.storyType
+    );
 
-  return (
-    <div className="ProjectBoard">
-      <SprintVelocitySimulation/>
-      <StorySearch projectId={projectId} loading={projectBoard.search.loading} />
-
-    <SideBar data-id="side-bar" buttons={sideBarButtons} />
-
-    <Notifications
-      notifications={notifications}
-      onRemove={removeNotification}
-      data-id="notifications"
-    />
-
-    <Columns {...columnProps} canCloseColumn={canCloseColumn(projectBoard)} />
-
-    <SearchResults />
-
-    {
-      history.status !== historyStatus.DISABLED &&
-      <Column
-        onClose={closeHistory}
-        title={`${I18n.t('projects.show.history')} '${history.storyTitle}'`}
-        data-id="history-column"
-      >
-        { history.status === historyStatus.LOADED
-          ? <History history={history.activities} data-id="history" />
-          : <ProjectLoading data-id="project-loading" />
-        }
-      </Column>
+    const newStories = moveTask(
+      sourceArray,
+      destinationArray,
+      sourceIndex,
+      destinationIndex
+    );
+
+    // Changing the column array order
+    if (dropColumn === "chillyBin") {
+      setNewChillyBinStories(newStories);
+    }
+
+    if (dropColumn === "backlog") {
+      setNewBacklogSprints(
+        getNewSprints(newStories, newBacklogSprints, sprintDropIndex)
+      );
     }
-    </div>
+    // Persisting the new array order
+    const newState = getNewState(isSameColumn, dropColumn, dragStory.state);
+    return dragDropStory(dragStory.id, dragStory.projectId, {
+      position: newPosition,
+      state: newState
+    });
+  };
+
+  if (!projectBoard.isFetched) {
+    return <b>{I18n.t("loading")}</b>;
```

Comment:
Use `<ProjectLoading data-id="project-loading" />` instead of `<b>{I18n.t("loading")}</b>;`